src/bindings/python/tests/test_graph/test_custom_op.py

```diff
@@ -76,6 +81,71 @@ def create_add_model():
     return Model(res, [param1, param2], "AddModel")
 
 
+class CustomOpWithAttribute(Op):
+    class_type_info = DiscreteTypeInfo("CustomOpWithAttribute", "extension")
+
+    def __init__(self, inputs, attrs):
+        super().__init__(self)
+        self._attrs = attrs
+        self.set_arguments(inputs)
+        self.constructor_validate_and_infer_types()
+
+    def validate_and_infer_types(self):
+        self.set_output_type(0, self.get_input_element_type(0), self.get_input_partial_shape(0))
+
+    def clone_with_new_inputs(self, new_inputs):
+        return CustomOpWithAttribute(new_inputs)
+
+    def get_type_info(self):
+        return CustomOpWithAttribute.class_type_info
+
+    def visit_attributes(self, visitor):
+        visitor.on_attributes(self._attrs)
+        return True
+
+
+@pytest.mark.parametrize(("attributes", "expectation", "raise_msg"), [
+    ({"axis": 0}, does_not_raise(), ""),
+    ({"value_str": "test_attribute"}, does_not_raise(), ""),
+    ({"value_float": 0.25}, does_not_raise(), ""),
+    ({"value_bool": True}, does_not_raise(), ""),
+    ({"list_str": ["one", "two"]}, does_not_raise(), ""),
+    ({"list_int": [1, 2]}, does_not_raise(), ""),
+    ({"list_float": np.array([1.5, 2.5], dtype="float32")}, does_not_raise(), ""),
+    ({"axis": 0, "list_int": [1, 2]}, does_not_raise(), ""),
+    ({"body": Model(ops.constant([1]), [])}, does_not_raise(), ""),
+    ({"dim": Dimension(2)}, does_not_raise(), ""),
+    ({"wrong_list": [{}]}, pytest.raises(TypeError), "Unsupported attribute type in provided list: <class 'dict'>"),
+    ({"wrong_np": np.array([1.5, 2.5], dtype="complex128")}, pytest.raises(TypeError), "Unsupported NumPy array dtype: complex128"),
+    ({"wrong": {}}, pytest.raises(TypeError), "Unsupported attribute type: <class 'dict'>")
+])
+def test_visit_attributes_custom_op(request, tmp_path, attributes, expectation, raise_msg):
+    input_shape = [2, 1]
+
+    param1 = ops.parameter(Shape(input_shape), dtype=np.float32, name="data1")
+    param2 = ops.parameter(Shape(input_shape), dtype=np.float32, name="data2")
+    custom = CustomOpWithAttribute(inputs=[param1, param2], attrs=attributes)
+    res = ops.result(custom, name="result")
+    model_with_op_attr = Model(res, [param1, param2], "CustomModel")
+
+    xml_path, bin_path = create_filename_for_test(request.node.name, tmp_path)
+
+    attrs = {}
+
+    with expectation as e:
+        serialize(model_with_op_attr, xml_path, bin_path)
+        ordered_ops = model_with_op_attr.get_ordered_ops()
+        ops_dict = {op.get_type_name(): op for op in ordered_ops}
+        attrs = ops_dict["CustomOpWithAttribute"].get_attributes()
+
+    if e is not None:
+        assert raise_msg in str(e.value)
+
+    for key, value in attrs.items():
+        assert key in attributes
+        assert attributes[key] == value
```

Comment:
```suggestion
    with expectation as e:
        serialize(model_with_op_attr, xml_path, bin_path)
        ordered_ops = model_with_op_attr.get_ordered_ops()
        ops_dict = {op.get_type_name(): op for op in ordered_ops}
        attrs = ops_dict["CustomOpWithAttribute"].get_attributes()
        for key, value in attrs.items():
            assert key in attributes
            assert attributes[key] == value

    if e is not None:
        assert raise_msg in str(e.value)
```

or use try ..catch e.g
```
try:
    # do test with test assertion
catch:
   # print error
finally:
   # test clean-up e.g remove created files
```

Reply:
The [pytest-style](https://docs.pytest.org/en/stable/reference/reference.html#pytest.raises) checking exceptions is using with-expression